matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/EventRelationsAggregationProcessor.kt

```diff
@@ -233,30 +239,35 @@ internal class EventRelationsAggregationProcessor @Inject constructor(
                         }
                     }
                 }
-                in EventType.POLL_START           -> {
+                in EventType.POLL_START             -> {
                     val content: MessagePollContent? = event.content.toModel()
                     if (content?.relatesTo?.type == RelationType.REPLACE) {
                         Timber.v("###REPLACE in room $roomId for event ${event.eventId}")
                         // A replace!
                         handleReplace(realm, event, content, roomId, isLocalEcho)
                     }
                 }
-                in EventType.POLL_RESPONSE        -> {
+                in EventType.POLL_RESPONSE          -> {
                     event.content.toModel<MessagePollResponseContent>(catchError = true)?.let {
                         handleResponse(realm, event, it, roomId, isLocalEcho)
                     }
                 }
-                in EventType.POLL_END             -> {
+                in EventType.POLL_END               -> {
                     event.content.toModel<MessageEndPollContent>(catchError = true)?.let {
                         handleEndPoll(realm, event, it, roomId, isLocalEcho)
                     }
                 }
-                in EventType.BEACON_LOCATION_DATA -> {
+                in EventType.STATE_ROOM_BEACON_INFO -> {
+                    event.content.toModel<LiveLocationBeaconContent>(catchError = true)?.let {
+                        liveLocationAggregationProcessor.handleBeaconInfo(realm, event, it, roomId, isLocalEcho)
+                    }
+                }
+                in EventType.BEACON_LOCATION_DATA   -> {
```

Comment:
I understand it may be confusing. What I suggest is trying to follow Matrix specifications:

- rename `LiveLocationBeaconContent` into `MessageBeaconInfoContent` which is the content of the state event of type `EventType.STATE_ROOM_BEACON_INFO`
- rename `MessageLiveLocationContent` into `MessageBeaconLocationDataContent` which is the content of the event of type `EventType.BEACON_LOCATION_DATA`
- rename `LiveLocationAggregatedSummary` into `LiveLocationShareAggregatedSummary` and `LiveLocationAggregatedSummaryEntity` into  `LiveLocationShareAggregatedSummaryEntity`

Does that make things clearer?

In terms of events, we first have a state event which is beacon info (it contains some info about the live) and then we have normal related events as beacons containing the location data.

Reply:
that sounds great to me! thanks for explaining

Reply:
Maybe move your very valuable comment to the codebase somewhere, where it will be more visible to other developer when looking in the area.

For instance `content of the state event of type EventType.STATE_ROOM_BEACON_INFO` could be in the Javadoc of `MessageBeaconInfoContent`

Reply:
Totally aggree, I added comments on both message contents.